components/backup-stream/tests/failpoints/mod.rs

```diff
@@ -400,6 +404,67 @@ mod all {
         )
     }
 
+    #[test]
+    fn encryption() {
+        let key_folder = TempDir::new().unwrap();
+        let key_file = key_folder.path().join("key.txt");
+        fail::cfg(
+            "log_backup::FileCore::should_swap_out::return_true",
+            "return()",
+        )
+        .unwrap();
+        std::fs::write(&key_file, "42".repeat(32) + "\n").unwrap();
+
+        let mut suite = SuiteBuilder::new_named("encryption")
+            .nodes(1)
+            .cluster_cfg(move |cfg| {
+                cfg.tikv.security.encryption.data_encryption_method = EncryptionMethod::Aes256Ctr;
+                cfg.tikv.security.encryption.master_key = MasterKeyConfig::File {
+                    config: FileConfig {
+                        path: key_file
+                            .to_str()
+                            .expect("cannot convert OsStr to Rust string")
+                            .to_owned(),
+                    },
+                };
+            })
+            .cfg(|cfg| cfg.temp_file_memory_quota = ReadableSize(16))
+            .build();
+
+        suite.must_register_task(1, "encryption");
+        let items = run_async_test(suite.write_records(0, 128, 1));
+        // So the old files can be "flushed" to disk.
+        let items2 = run_async_test(suite.write_records(256, 128, 1));
+        suite.sync();
+
+        let files = WalkDir::new(suite.temp_files.path())
+            .into_iter()
+            .filter_map(|file| file.ok().filter(|f| f.file_type().is_file()))
+            .collect::<Vec<_>>();
+        assert!(!files.is_empty());
+        for dir in files {
+            let data = std::fs::read(dir.path()).unwrap();
+            // assert it contains data...
+            assert_ne!(data.len(), 0);
+            // ... and is plain zstd compression.
+            assert_ne!(
```

Comment:
This is just a not equal assertion, can't prove the data is encrypt as planned.

Reply:
Yes, it only makes sure that the content was probably encrypted. Because the CTR mode have a random initial vector, it is a little bit harder to verify by an equal check.

Reply:
(I have added more assertions to make sure there isn't values presented in plain text.)